pkg/ccl/backupccl/restore_data_processor.go

```diff
@@ -209,23 +225,90 @@ func inputReader(
 	}
 }
 
-func (rd *restoreDataProcessor) runRestoreWorkers(entries chan execinfrapb.RestoreSpanEntry) error {
+type mergedSST struct {
+	entry   execinfrapb.RestoreSpanEntry
+	iter    storage.SimpleMVCCIterator
+	cleanup func()
+}
+
+func (rd *restoreDataProcessor) openSSTs(
+	entry execinfrapb.RestoreSpanEntry, sstCh chan mergedSST,
+) error {
+	ctx := rd.Ctx
+
+	// The sstables only contain MVCC data and no intents, so using an MVCC
+	// iterator is sufficient.
+	var iters []storage.SimpleMVCCIterator
+	var dirs []cloud.ExternalStorage
+
+	// sendIters sends all of the currently accumulated iterators over the
+	// channel.
+	sendIters := func(itersToSend []storage.SimpleMVCCIterator, dirsToSend []cloud.ExternalStorage) {
+		multiIter := storage.MakeMultiIterator(itersToSend)
+
+		cleanup := func() {
+			for _, iter := range itersToSend {
+				iter.Close()
+			}
+			multiIter.Close()
```

Comment:
is this still swapped? should we be closing the multi before its underlying iters?

Reply:
Yep, thank you.